internal/handlers/sqlite/msg_createindexes.go

```diff
@@ -401,3 +436,106 @@ func processIndexKey(command string, keyDoc *types.Document) ([]backends.IndexKe
 		})
 	}
 }
+
+// formatIndexKey formats the given index key to a string.
+func formatIndexKey(key []backends.IndexKeyPair) string {
+	res := make([]string, len(key))
+
+	for i, pair := range key {
+		order := "1"
+		if pair.Descending {
+			order = "-1"
+		}
+
+		res[i] = pair.Field + ": " + order
+	}
+
+	return strings.Join(res, ", ")
+}
+
+// validateIndexesForCreation validates the given list of indexes to create against the existing ones.
+func validateIndexesForCreation(command string, existing, toCreate []backends.IndexInfo) error {
+	for i, newIdx := range toCreate {
+		newKey := formatIndexKey(newIdx.Key)
+
+		if newIdx.Name == "" {
+			msg := fmt.Sprintf(
+				"Error in specification { key: { %s }, name: %q, v: 2 } :: caused by :: index name cannot be empty",
+				newKey, newIdx.Name,
+			)
+
+			return commonerrors.NewCommandErrorMsgWithArgument(commonerrors.ErrCannotCreateIndex, msg, command)
+		}
+
+		if newIdx.Name == backends.DefaultIndexName && newKey != "_id: 1" {
+			msg := fmt.Sprintf(
+				"The index name '_id_' is reserved for the _id index, which must have key pattern {_id: 1},"+
+					" found key: { %s }",
+				newKey,
+			)
+
+			return commonerrors.NewCommandErrorMsgWithArgument(commonerrors.ErrBadValue, msg, command)
+		}
+
+		// Check for duplicates within the list of indexes to create.
+		for j := i - 1; j >= 0; j-- {
+			prevKey := formatIndexKey(toCreate[j].Key)
+			prevName := toCreate[j].Name
+
+			if prevName == newIdx.Name && prevKey == newKey {
+				msg := fmt.Sprintf("Identical index already exists: %s", prevName)
+
+				return commonerrors.NewCommandErrorMsgWithArgument(commonerrors.ErrIndexAlreadyExists, msg, command)
+			}
+
+			if newIdx.Name == prevName {
+				msg := fmt.Sprintf(
+					"An existing index has the same name as the requested index."+
+						" When index names are not specified, they are auto generated and can cause conflicts."+
+						" Please refer to our documentation. Requested index: { key: { %s }, name: %q },"+
+						" existing index: { key: { %s }, name: %q }",
+					newKey, newIdx.Name, prevKey, prevName,
+				)
+
+				return commonerrors.NewCommandErrorMsgWithArgument(commonerrors.ErrIndexKeySpecsConflict, msg, command)
+			}
+
+			if newKey == prevKey {
+				msg := fmt.Sprintf(
+					"Index already exists with a different name: %s", prevName,
+				)
+
+				return commonerrors.NewCommandErrorMsgWithArgument(commonerrors.ErrIndexOptionsConflict, msg, command)
+			}
+		}
+
+		// Check for conflicts with existing indexes.
+		for _, existingIdx := range existing {
+			existingKey := formatIndexKey(existingIdx.Key)
+
+			if newIdx.Name == existingIdx.Name && newKey == existingKey {
+				// Fully identical indexes are allowed.
+				break
```

Comment:
Hmm, I don't think so. We already found a fully identical index, and it's ok. Partially identical indexes are not possible, because in `existing` there is that fully identical index that we found, and it's not possible that this index is in conflict with some other index in `existsing`.